tools/link_system_libs

```diff
@@ -19,82 +20,128 @@
 
 
-# This is a shell script instead of a python script to allow for
-# easily calling both the system python3 (without the pipenv) and the
-# pipenv python3 (inside the pipenv).
+import subprocess
 
-# Non-sh bash features used:
-#
-#   * array variables
-#
-#     This makes the support for systems with python site-packages in
-#     more than one lib dir, e.g. in 'lib' and 'lib64', much easier.
-#
-#   * [[ ]]
-#
-#     This is more or less due to programmer laziness.
-
-
-# The libs to symlink from the system into the pipenv
-LIBS=("gi")
-
-# The system python (as opposed to the pipenv python)
-SYSPYTHON="/usr/bin/python3"
-
-
-libpaths() {
-    local PYTHON=$1
-    $PYTHON -c "import sys; print('\n'.join(sys.path))" | grep -e "site-packages" -e "dist-packages" | sort
-}
-
-syspaths=($(libpaths $SYSPYTHON))
-echo "syspaths:" "${syspaths[@]}"
-
-envpaths=($(libpaths "pipenv run python3"))
-echo "envpaths:" "${envpaths[@]}"
-
-if [[ "${#syspaths[@]}" -ne "${#envpaths[@]}" ]]; then
-    echo "Path array size mismatch"
-    exit 1  # path array size should the same
-fi
-
-for i in "${!syspaths[@]}"; do
-    envpath="${envpaths[$i]}"
-    syspath="${syspaths[$i]}"
-    echo
-    echo "envpath: $envpath"
-    echo "syspath: $syspath"
-
-    for lib in "${LIBS[@]}"; do
-	linkpath="$envpath/$lib"
-	linktarget="$syspath/$lib"
-	echo
-	echo "linkpath: $linkpath"
-	echo "linktarget: $linktarget"
-	if test -L "$linkpath" && test "x$(readlink "$linkpath")" = "x$linktarget"; then
-	    echo "Nothing to do ($linkpath already pointing to $linktarget)"
-	    continue  # nothing to do: link exists and points to correct target
-	fi
-	if test -e "$linkpath"; then
-	    echo "Error: Link $linkpath already exists"
-	    exit 1  # link already exists and does not point to correct target
-	fi
-	if test -e "$linktarget"; then
-	    echo "Link target does not exist, skipping"
-	    continue  # link target does not exist
-	fi
-
-	echo "Linking $linktarget to $linkpath"
-	if ln -s "$linktarget" "$linkpath"; then
-	    echo "Linking successful"
-	else
-	    echo "Linking has failed."
-	    exit 1  # error while creating the link
-	fi
-    done
-done
-
-echo
-echo "Finished linking system libs."
+from pathlib import Path
+
+
+class Symlinker:
+    def __init__(self):
+        self.system_site_packages = self.get_site_packages(["/usr/bin/python3"])
+        self.pipenv_site_packages = self.get_site_packages(["pipenv", "run", "python3"])
+
+    def get_site_packages(self, python_cmd):
+        """Get the given python_cmd's sys.path elements ending in '/site-packages' (or '/dist-packages' for Debian)"""
+
+        cmd = python_cmd + ["-c", "import sys; print('path', '=', repr(sys.path))"]
+        proc = subprocess.run(cmd, check=True, capture_output=True, text=True,)
+        vars = {}
+        exec(proc.stdout, vars, vars)
+        path = [Path(elt) for elt in vars["path"]]
+        return [elt for elt in path if elt.name == ["site-packages", "dist-packages"]]
```

Comment:
```suggestion
        return [elt for elt in path if elt.name in ["site-packages", "dist-packages"]]
```

Reply:
Fix in [`session-dbus-wip`](https://github.com/ndim/socranop/tree/session-dbus-wip). Sqash into c0bdea24f00afd9c65e33c9f7e6387d2e5b91b8b or commit on top of [`session-dbus`](https://github.com/ndim/socranop/tree/session-dbus)?

Reply:
Squash wherever you're comfortable.  With the original commit is cleaner, but if it's painful don't worry too much about it.

Reply:
Added onto session-dbus-fixes and squashed into session-dbus-fixed, choose your preferred branch.

Reply:
In session-dbus.